execution/evm/vm.go

```diff
@@ -897,6 +923,8 @@ func (vm *VM) call(caller acm.Account, callee acm.MutableAccount, code, input []
 					return nil, firstErr(err, popErr)
 				}
 			}
+
+			vm.returnData = nil
```

Comment:
here too, I just think it's marginally better immediately once we enter this case block.